beatrix/src/test/java/org/killbill/billing/beatrix/integration/TestWithUsagePlugin.java

```diff
@@ -187,17 +190,58 @@ public void testSimple() throws Exception {
 
     }
 
+
+    @Test(groups = "slow")
+    public void testDryRunWithPluginProperties() throws Exception {
+
+        clock.setDay(new LocalDate(2023, 6, 1));
+
+        final AccountData accountData = getAccountData(1);
+        final Account account = createAccountWithNonOsgiPaymentMethod(accountData);
+        accountChecker.checkAccount(account.getId(), accountData, callContext);
+
+        final DefaultEntitlement bpSubscription = createBaseEntitlementAndCheckForCompletion(account.getId(), "bundleKey", "Shotgun", ProductCategory.BASE, BillingPeriod.ANNUAL, NextEvent.CREATE, NextEvent.BLOCK, NextEvent.INVOICE);
+        // Check bundle after BP got created otherwise we get an error from auditApi.
+        subscriptionChecker.checkSubscriptionCreated(bpSubscription.getId(), internalCallContext);
+        invoiceChecker.checkInvoice(account.getId(), 1, callContext, new ExpectedInvoiceItemCheck(new LocalDate(2023, 6, 1), null, InvoiceItemType.FIXED, new BigDecimal("0")));
+
+        //
+        // ADD ADD_ON ON THE SAME DAY
+        //
+        final DefaultEntitlement aoSubscription = addAOEntitlementAndCheckForCompletion(bpSubscription.getBundleId(), "Bullets", ProductCategory.ADD_ON, BillingPeriod.NO_BILLING_PERIOD, NextEvent.CREATE, NextEvent.BLOCK, NextEvent.NULL_INVOICE);
+
+        testUsagePluginApi.recordUsageData(aoSubscription.getId(), "tracking-1", "bullets", new DateTime(2023, 6, 1, 13, 25, 14, DateTimeZone.UTC), BigDecimal.valueOf(150), callContext);
+
+        final List<PluginProperty> pluginProperties = new ArrayList<>();
+        pluginProperties.add(new PluginProperty("foo", "bar", true));
+        pluginProperties.add(new PluginProperty("lily", "lolo", true));
+        pluginProperties.add(new PluginProperty("", "", true));
+
+        // We expect to see those 3 + 2 (DRY_RUN_CUR_DATE, DRY_RUN_TARGET_DATE) set by the system
+        testUsagePluginApi.setExpectedPluginProperties(5);
+
+        invoiceUserApi.triggerDryRunInvoiceGeneration(account.getId(), new LocalDate(2023, 7, 1), new TestDryRunArguments(DryRunType.TARGET_DATE), pluginProperties, callContext);
+
+    }
+
     public static class TestUsagePluginApi implements UsagePluginApi {
 
         private final SortedMap<DateTime, List<RawUsageRecord>> usageData;
+        private int expectedPluginProperties;
 
         public TestUsagePluginApi() {
             this.usageData = new TreeMap<>();
+            this.expectedPluginProperties = 0;
         }
 
         @Override
         public List<RawUsageRecord> getUsageForAccount(final DateTime startDate, final DateTime endDate, final UsageContext usageContext, final Iterable<PluginProperty> properties) {
 
+            final int nbProps = Iterables.size(properties);
+            if (nbProps != expectedPluginProperties) {
```

Comment:
Could have just used `assertEquals`? 😄

Reply:
Indeed... Fixed in https://github.com/killbill/killbill/pull/1885/commits/cfc01e2ea9c973e1b1b38938ee55ae39e78706c2